src/main/java/chess/dao/ChessService.java

```diff
@@ -0,0 +1,99 @@
+package chess.dao;
+
+import chess.domain.Board;
+import chess.domain.ChessGame;
+import chess.domain.Movement;
+import chess.domain.piece.abstractPiece.Piece;
+import chess.domain.piece.character.Team;
+import chess.exception.ConnectionException;
+import chess.exception.DataAccessException;
+import chess.exception.InvalidGameRoomException;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class ChessService {
+    private final ConnectionGenerator connectionGenerator;
+    private final PiecesDao piecesDao;
+    private final ChessGameDao chessGameDao;
+
+    public ChessService() {
+        this(new ConnectionGenerator(), new PiecesDao(), new ChessGameDao());
+    }
+
+    private ChessService(
+            ConnectionGenerator connectionGenerator,
+            PiecesDao piecesDao,
+            ChessGameDao chessGameDao
+    ) {
+        this.connectionGenerator = connectionGenerator;
+        this.piecesDao = piecesDao;
+        this.chessGameDao = chessGameDao;
+    }
+
+    public void initialize(Board board, Team team, String roomName) {
+        try (final Connection connection = connectionGenerator.getConnection()) {
+            connection.setAutoCommit(false);
+            try {
+                chessGameDao.add(team, roomName, connection);
+                piecesDao.addAll(board, roomName, connection);
+            } catch (InvalidGameRoomException | DataAccessException e) {
+                connection.rollback();
+                throw e;
+            }
+            connection.commit();
+        } catch (SQLException e) {
+            throw new ConnectionException(e.getMessage());
+        }
+    }
+
+    public ChessGame loadChessGame(String roomName) {
+        try (final Connection connection = connectionGenerator.getConnection()) {
+            connection.setAutoCommit(false);
+            Team currentTeam;
+            Board board;
+            try {
+                currentTeam = chessGameDao.findCurrentTeamByRoomName(roomName, connection);
+                board = piecesDao.loadAll(roomName, connection);
+            } catch (InvalidGameRoomException | DataAccessException e) {
+                connection.rollback();
+                throw e;
+            }
```

Comment:
```suggestion
            } catch (Exception e) {
                connection.rollback();
                throw e;
            }
```
정의하지 않은 예외가 발생해도 롤백을 해줘야하기 때문에 
모든 Exception을 잡도록 해주는게 좋을것 같네요!

Reply:
맞는 말씀이네요! RuntimeException과 SQLException 모두 잡도록 변경했습니다